client/app/bundles/course/survey/containers/PublishToggle.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import Toggle from 'material-ui/Toggle';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { setPublishedAttribute } from '../actions/surveys';
+import surveyTranslations from '../translations';
+import { surveyShape } from '../propTypes';
+
+const translations = defineMessages({
+  publishSuccess: {
+    id: 'course.surveys.PublishToggle.publishSuccess',
+    defaultMessage: 'Survey "{title}" published.',
+  },
+  publishFailure: {
+    id: 'course.surveys.PublishToggle.publishFailure',
+    defaultMessage: 'Failed to publish "{title}".',
+  },
+  unpublishSuccess: {
+    id: 'course.surveys.PublishToggle.unpublishSuccess',
+    defaultMessage: 'Survey "{title}" unpublished.',
+  },
+  unpublishFailure: {
+    id: 'course.surveys.PublishToggle.unpublishFailure',
+    defaultMessage: 'Failed to unpublish "{title}".',
+  },
+});
+
+class PublishToggle extends React.Component {
+  handleToggle = (event, value) => {
+    const { dispatch, survey, courseId, surveyId } = this.props;
+    const successTranslation =
+      value ? translations.publishSuccess : translations.unpublishSuccess;
+    const failureTranslation =
+      value ? translations.publishFailure : translations.unpublishFailure;
+
+    dispatch(setPublishedAttribute(
+      courseId,
+      surveyId,
+      value,
+      <FormattedMessage {...successTranslation} values={survey} />,
+      <FormattedMessage {...failureTranslation} values={survey} />
+    ));
+  }
+
+  render() {
+    const { survey, noLabel } = this.props;
+
+    if (!survey.canUpdate) {
+      return null;
+    }
+
+    return (
+      <Toggle
+        label={noLabel ? null : <FormattedMessage {...surveyTranslations.published} />}
+        labelPosition="right"
+        toggled={survey.published}
+        onToggle={this.handleToggle}
+      />
+    );
+  }
+}
+
+PublishToggle.propTypes = {
+  dispatch: PropTypes.func.isRequired,
+  courseId: PropTypes.string.isRequired,
+  surveyId: PropTypes.string.isRequired,
+  noLabel: PropTypes.bool,
```

Comment:
I thought it would be better use of screen space to put the toggle on the survey index page instead of using the 'ban' sign. Will add a commit to this PR to do that.